docs/DeveloperGuide.md

```diff
@@ -2,32 +2,285 @@
+#### How the feature is implemented:
+* The `ListSchoolCommand` class extends `Command` class where it overrides the `execute` method for
+  custom behaviour.
+* The command first reads a JSON file to obtain the names via `createJsonObject()` method from the
+  superclass.
+* The `displaySchoolList()` method will iterate over the keys of the database which contains the University
+  names, upon acquiring the keys, they will be printed over the CLI.
+* There are also assertions and logging in place for error handling.
+* Line Separator is used to ensure readability and ease of use for users.  
+
+#### Why it is implemented that way:
+* The `execute` method is essential and unique to every command class so inheritance was used. 
+* Every method in the class remains maintainable and has one responsibility this allows easy debugging and
+  refactoring.
+* By using inheritance, new command classes can easily extend the functionality of existing ones
+  which reducing redundancy in the code
+* Logging and assertions helps the team of developers to follow through the command execution.
+
+#### Alternatives considered:
+* Reading of the `database.json` was tricky and other libraries were considered.
+* Considered placing all the class methods inside the `execute` method but kept SLAP in mind to ensure 
+  readability.
+
+#### Sequence Diagram on PlantUML:
+![List School Command Sequence Diagram](images/ListSchoolsCommand.png)
+
+### 2. Filter Courses Command
+
+#### Overview:
+This command is responsible for displaying and retrieving the full list of mappable courses from the partner 
+universities to a user specified NUS course from `database.json` file. It helps the users to identify whether
+that NUS course is suitable to be mapped overseas in South East Asia and Oceania.
+
+#### How the feature is implemented:
+* The `FilterCoursesCommand` class extends `Command` class where it overrides the `execute` method for
+  custom behaviour.
+* The command first reads a JSON file to obtain the names via `createJsonObject()` method from the
+  superclass.
+* The `getNusCourseCode` method then extract out the user specified NUS course code from the input, which is in the
+  format: `filter COURSE_CODE`.
+* The NUS course code is then passed into the `displayMappableCourses()` method along with teh Json object. The method
+  will iterate over the keys of the database which contains the University names, then obtain the array `courses`
+  stored in the "courses" field. The `courses` array is then iterated over, for each course,
+  if the value in the "nus_course_code" is equals to the NUS course code passed into the method, the university name and
+  "pu_course_code" value of the course will be printed to the CLI.
+* There are also assertions and logging in place for error handling.
+* Line Separator is used to ensure readability and ease of use for users.
+
+#### Sequence Diagram on PlantUML:
+![Filter Courses Sequence Diagram](images/FilterCoursesCommand.png)
```

Comment:
Should there be a third activation bar for "print PU and PU course code"?

Reply:
Also should "throw new IOException()" be depicted as a solid black arrow within Command?